modules/apps/transfer/keeper/msg_server_test.go

```diff
@@ -125,3 +126,47 @@ func (suite *KeeperTestSuite) TestMsgTransfer() {
 		})
 	}
 }
+
+// TestUpdateParams tests UpdateParams rpc handler
+func (suite *KeeperTestSuite) TestUpdateParams() {
+	validAuthority := suite.chainA.GetSimApp().TransferKeeper.GetAuthority()
+	testCases := []struct {
+		name    string
+		msg     *types.MsgUpdateParams
+		expPass bool
+	}{
+		{
+			"success",
+			types.NewMsgUpdateParams(validAuthority, types.DefaultParams()),
+			true,
+		},
+		{
+			"malformed authority address",
+			types.NewMsgUpdateParams(ibctesting.InvalidID, types.DefaultParams()),
+			false,
+		},
+		{
+			"empty authority address",
+			types.NewMsgUpdateParams("", types.DefaultParams()),
+			false,
```

Comment:
maybe we should add a test for a string of whitespace `"           "` also failing here.

Reply:
done